src/main/java/edu/osu/waiting4ubackend/client/PetSearchQueryBuilder.java

```diff
@@ -0,0 +1,67 @@
+package edu.osu.waiting4ubackend.client;
+
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.EntityQuery;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.StructuredQuery;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static edu.osu.waiting4ubackend.client.PetDBClient.PETS_COLLECTION_NAME;
+
+public class PetSearchQueryBuilder {
+    private String type;
+    private String breed;
+    private List<String> dispositions;
+
+    public PetSearchQueryBuilder() {
+
+    }
+
+    public PetSearchQueryBuilder setType(String type) {
+        this.type = type;
+        return this;
+    }
+
+    public PetSearchQueryBuilder setBreed(String breed) {
+        this.breed = breed;
+        return this;
+    }
+
+    public PetSearchQueryBuilder setDispositions(List<String> dispositions) {
+        this.dispositions = dispositions;
+        return this;
+    }
+
+    public Query<Entity> build() {
+        List<StructuredQuery.Filter> list = new ArrayList<>();
+        if(type != null) {
+            list.add(StructuredQuery.PropertyFilter.eq("type", type));
+        }
+        if(breed != null) {
+            list.add(StructuredQuery.PropertyFilter.eq("breed", breed));
+        }
+        if(dispositions != null) {
```

Comment:
I am thinking if we can do a for loop instead of an if/else statement? The reason I am thinking about that is that maybe in the future, we may add more disposition options, and a for loop would handle multiple options easier. What do you think?

Reply:
That's a good suggestion! I will refactor it latter